game/theater/theatergroundobject.py

```diff
@@ -481,18 +488,64 @@ def __init__(
             dcs_identifier="AA",
             sea_object=False,
         )
-        # Set by the SAM unit generator if the generated group is compatible
-        # with Skynet.
-        self.skynet_capable = False
+        self.aa_ranges: List[AirDefenseRange] = [aa_range]
 
     @property
+    # Returns the TGO Name
     def group_name(self) -> str:
-        if self.skynet_capable:
-            # Prefix the group names of SAM sites with the side color so Skynet
-            # can find them.
-            return f"{self.faction_color}|SAM|{self.group_id}"
+        return super().group_name
+
+    # Returns the name of one single group
+    def groups_name(self, group: Group) -> str:
+        try:
+            group_index = self.groups.index(group)
+            if self.skynet_capability_for_range(
+                self.aa_ranges[group_index], (group_index > 0)
+            ):
+                return self.skynet_group_name(group_index)
+        except AttributeError:  # Allow backwards compatibility
+            return self.skynet_group_name()
+        except ValueError:  # Allow backwards compatibility
+            pass
+
+        return super().group_name
+
+    def set_groups(self, groups: List[Group], ranges: List[AirDefenseRange]):
+        self.groups = groups
+        self.aa_ranges = ranges
+
+    def skynet_capability_for_range(
```

Comment:
We have a lot more data than we did when this was first written. I assume you're trying to capture anything radar capable? Can maybe use `game/data/radar_db.py` for this?

I think that helps fix another bug here too (a pre-existing one): I'm pretty sure we need to avoid adding groups to skynet if they have lost their radars.

Reply:
Good idea! this is an addition.
Primarily the idea here is to stick to the manual of skynet and only add mid and long range sams to the skynet. Also add SHORAD if they act as Point defence so that they wont go dark if the attached SAM will defend harm by going dark.

But your addition is also perfect here. i reused the has_live_radar_sam as well as checking if the first unit of the group is a Radar Unit (which is also a requirement from skynet to work properly). I also cross checked every sam generator that SR is always the first unit.

Reply:
> I'm pretty sure we need to avoid adding groups to skynet if they have lost their radars.

Based on https://github.com/dcs-liberation/dcs_liberation/issues/405 I think I'm remembering incorrectly. Sorry I didn't think of that sooner!